### PR TITLE
web: make sail-info its own component, show "connect to sail" button if sail enabled and no url

### PR DESCRIPTION
**Pull request by maiamcc**
Hello @yuindustries, @nicks,

Please review the following commits I made in branch maiamcc/sail-info-component:

05454210c972ade589254606635e38c555f97545 (2019-04-26 13:36:28 -0400)
clean up go

b8e711c168237345b5d485d1b670e6dc20571ec6 (2019-04-26 13:35:36 -0400)
update tests

04bb99e5981e52bbd7b1edb89109a268db702d91 (2019-04-26 13:09:55 -0400)
sailinfo is its own component

b463de5c385e48f33be49d515c8c78dbd7e86b7f (2019-04-26 11:24:32 -0400)
share button in ui if sail enabled but no url

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: web/src/SailInfo.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.SailInfo {
+  padding: 1em;
```
Comment (yuindustries) [inline web/src/SailInfo.scss]: Use $spacing-unit from constants whenever possible

File: web/src/HUD.tsx [hunk 2]
```diff
@@ -56,6 +56,7 @@ type HudState = {
     Resources: Array<Resource>
     Log: string
     LogTimestamps: boolean
+    SailEnabled: boolean
```
Comment (maiamcc) [inline web/src/HUD.tsx]: idk if this is really the right place to put this info? but i can't think of a better one 🤷‍♀️

File: internal/cli/up.go [hunk 3]
```diff
@@ -147,7 +147,7 @@ func (c *upCmd) run(ctx context.Context, args []string) error {
 
 	g.Go(func() error {
 		defer cancel()
-		return upper.Start(ctx, args, c.watch, triggerMode, c.fileName, c.hud)
```
Comment (nicks) [inline internal/cli/up.go]: it might be easier to dependency inject this with wire, so you don't need to wire it all up yourself?

Comment (maiamcc) [reply internal/cli/up.go]: huhhh wiring this straight into `NewStore()` is preferable? idk how to get it onto the store via wire otherwise...

**Pull request merged**